### PR TITLE
Minor cleanups of merge_leaf_stats()

### PR DESCRIPTION
**Pull request by danielgustafsson**
Two small cleanups in `merge_leaf_stats()` noticed when passing through. See the individual commits for details, commit messages pasted below:

Scribbling on unused parameters rather than allocating local variables isn't advised behavior even when it's safe to do. Move to using our own variables instead.

Elevel LOG should be treated with care, as it quickly can bloat the logfile drowning out more important messaging. This message doesn't seem like a LOG event, but rather a DEBUG2 event.

### CODE REVIEW

File: src/backend/commands/analyze.c [hunk 1]
```diff
@@ -3823,7 +3826,7 @@ merge_leaf_stats(VacAttrStatsP stats,
 	float nmultiple = 0; // number of values that appeared more than once
 	bool allDistinct = false;
 	int slot_idx = 0;
-	samplerows = 0;
+	int sampleRows = 0;
```
Comment (vraghavan78) [inline src/backend/commands/analyze.c]: can we change the name too? `sampleCount` else similar to input parameter

Comment (danielgustafsson) [reply src/backend/commands/analyze.c]: Yep, no worries, I can rename it to `sampleCount` before pushing.

**Comment by hlinnaka** [comment]
The original HLL commit added a lot of LOG messages. I've been wondering about these for a while, because I'm seeing the "Computing Stats for column" messages in the log every time I run initdb, but haven't gotten around to complaining :-).

```
$ git log -p -n1 9c1b1ae3afe999031f2e43a87f7dbae61b9c7fbe | grep LOG,
+			elog (LOG,"HLL FULL SCAN ");
+		elog (LOG,"Needs sample for %s " , RelationGetRelationName(onerel));
+	elog(LOG, "Computing Minimal Stats column %d", stats->attr->attnum);
+	elog(LOG, "Computing Very Minimal Stats for  column %d", stats->attr->attnum);
+	elog(LOG, "Computing Scalar Stats  column %d", stats->attr->attnum);
+	elog(LOG, "Merging leaf stats");
```

I'd suggest demoting all of them.

File: src/backend/commands/analyze.c (hunk 1 above)
Comment (vraghavan78) [reply src/backend/commands/analyze.c]: Thanks!

**Comment by danielgustafsson** [comment]
> The original HLL commit added a lot of LOG messages.

I did notice all the `LOG` entries when removing the demoted lines from the expected regress file, but  was too tired to look into it at the time. 

Digging around a little, it seems it was done very much on purpose. Commit e725d5af3b310cdefd5c5585b38936bfd56c1b91 is a good example. @oarap @bhuvnesh2703 what was the rationale behind the verbose logging, I fail to see a scenario when pushing this much into the logs will be helpful to the user?

I agree with demoting these to DEBUG2, but I will open a separate PR for that since the scope increased from this.

**Comment by bhuvnesh2703** [comment]
@danielgustafsson  i remember having a discussion with the same concern about these log messaging bloating the logs, but we ended up doing it. I think there was some customer request or something.

**Comment by vraghavan78** [comment]
@bhuvnesh2703 I donot a customer asked for this particular things. I agree with @danielgustafsson on not wasting another GUC on this.

**Comment by bhuvnesh2703** [comment]
@vraghavan78 @danielgustafsson cool then go ahead.

**Comment by danielgustafsson** [comment]
> On 14 Nov 2018, at 22:40, Bhuvnesh <notifications@github.com> wrote:
> 
> @vraghavan78 @danielgustafsson cool then go ahead.

Thanks for clarifying, will go ahead with the LOG messages tomorrow and will open a new PR for those.

**Comment by danielgustafsson** [comment]
Variable handling commit pushed to master and the logging patch moved to #6277 where it has been extended handling more LOG->DEBUG2.

**Pull request closed without merging**